RealmWeaver/ModuleWeaver.cs

```diff
@@ -304,28 +302,55 @@ private void WeaveType(TypeDefinition type, Dictionary<string, Tuple<MethodRefer
-        else if (prop.PropertyType.Name == "RealmList`1" && prop.PropertyType.Namespace == "Realms")
-        {
-            // RealmList allows people to declare lists only of _realmObject due to the class definition
-            if (!prop.IsAutomatic())
-            {
+
+        // treat IList and RealmList similarly but IList gets a default so is useable as standalone
+        // IList or RealmList allows people to declare lists only of _realmObject due to the class definition
+        else if (prop.PropertyType.Name == "IList`1" && prop.PropertyType.Namespace == "System.Collections.Generic") {
+            var elementType = ((GenericInstanceType)prop.PropertyType).GenericArguments.Single();
+            if (!elementType.Resolve().BaseType.IsSameAs(_realmObject)) {
+                LogWarningPoint(
+  $"SKIPPING {type.Name}.{columnName} because it is an IList but its generic type is not a RealmObject subclass, so will not persist",
+  sequencePoint);
+                return;
+            }
+
+            if (prop.SetMethod != null) {
                 LogErrorPoint(
-                    $"{type.Name}.{columnName} is not an automatic property but its type is a RealmList which normally indicates a relationship",
+                    $"{type.Name}.{columnName} has a setter but its type is a IList which only supports getters",
                     sequencePoint);
                 return;
             }
-            if (prop.SetMethod != null)
+            var concreteListType = new GenericInstanceType(System_IList) { GenericArguments = { elementType } };
+            var listConstructor = concreteListType.Resolve().GetConstructors().Single(c => c.IsPublic && c.Parameters.Count == 0);
+            var concreteListConstructor = listConstructor.MakeHostInstanceGeneric(elementType);
+
+            // weaves list getter which also sets backing to List<T>, forcing it to accept us setting it post-init
+            var backingDef = backingField as FieldDefinition;
+            if (backingDef != null)
             {
+                backingDef.Attributes &= ~FieldAttributes.InitOnly;  // without a set; auto property has this flag we must clear
+            }
+            ReplaceListGetter(prop, backingField, columnName,
+                new GenericInstanceMethod(_genericGetListValueReference) { GenericArguments = { elementType } }, elementType,
+                             ModuleDefinition.ImportReference(concreteListConstructor) );
+        } else if (prop.PropertyType.Name == "RealmList`1" && prop.PropertyType.Namespace == "Realms") {
+            var elementType = ((GenericInstanceType)prop.PropertyType).GenericArguments.Single();
+            if (!elementType.Resolve().BaseType.IsSameAs(_realmObject)) {
+                LogWarningPoint(
+  $"SKIPPING {type.Name}.{columnName} because it is a RealmList but its generic type is not a RealmObject subclass, so will not persist",
```

Comment:
Compiler will allow a further descendant of `RealmObject` - there is no way to constrain to just **direct** descendants.